### ISSUE TITLE
Segfault when using invalid regular expression "["

### ISSUE THREAD

**Issue by louisjac** [neomutt/neomutt#2664]
## Expected Behaviour

I was trying to search mails starting with "[" but instead of typing the regular expression "[[]" I wrote "[".

## Actual Behaviour

Neomutt crashed with a segfault.

If NeoMutt crashed, did your OS create a 'coredump' file?

Nope.

## Steps to Reproduce

Using the regular expression [ to search or limit the view to messages matching the (invalid) regular expression "[".

## How often does this happen?

Always

## When did it start to happen?

I don't know but the original mutt refuses the regexp and does not crash.

## NeoMutt Version

```
NeoMutt 20200821
Copyright (C) 1996-2020 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.8.7-arch1-1 (x86_64)
ncurses: ncurses 6.2.20200212 (compiled with 6.2.20200212)
libidn2: 2.3.0 (compiled with 2.3.0)
GPGME: 1.14.0
GnuTLS: 3.6.14
libnotmuch: 5.2.0
storage: kyotocabinet, gdbm, lmdb
compression: lz4, zlib, zstd

Configure options: --prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --gpgme --sqlite --autocrypt --lua --notmuch --gss --gnutls --sasl --with-ui=ncurses --with-idn2=/usr --disable-idn --idn2 --lmdb --kyotocabinet --gdbm --lz4 --zlib --zstd

Compilation CFLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -std=c99 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -I/usr/include/lua5.3 -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo +gnutls +gpgme +gss +hcache -homespool +idn +inotify 
  -locales_hack +lua +meta -mixmaster +nls +notmuch -openssl +pgp +regex +sasl 
  +smime +sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"
```

## Extra Info

* Operating System and its version
 → Archlinux
* Were you using multiple copies of NeoMutt at once?
 → No
* Were you using 'screen' or 'tmux'?
 → No
* Is your email local (maildir) or remote (IMAP)?
→ local maildlir

**Comment by flatcap** [comment]
Sorry about that.
Good news though.  The bug is already fixed (in commit e25b8db91).
There's a new release due on Friday.

**Comment by louisjac** [comment]
Ah ok, I had not seen that. Thanks!
